### PR TITLE
chore(grpc): move metrics singletons to pkg/

### PR DESCRIPTION
**Pull request by RTann**
## Description

As part of the `scanner` [go module migration](https://github.com/stackrox/stackrox/pull/7042) into `rox`, scanner code cannot directly call functions defined under `central`. This PR moves the gRPC metrics endpoint into the `pkg/` directory, so it'll be visible to Scanner as well.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

If it builds, it builds

### CODE REVIEW

File: pkg/grpc/metrics/http_metrics.go [hunk 1]
```diff
@@ -13,9 +15,22 @@ type HTTPMetrics interface {
...
+
 // NewHTTPMetrics returns a new HTTPMetrics object
 func NewHTTPMetrics() HTTPMetrics {
 	return &httpMetricsImpl{
 		allMetrics: make(map[string]*perPathHTTPMetrics),
 	}
 }
+
+// HTTPSingleton returns a singleton of a HTTPMetrics struct
+func HTTPSingleton() HTTPMetrics {
```
Comment (connorgorman) [inline pkg/grpc/metrics/http_metrics.go:31]: This is an anti-pattern. Singletons should not be in pkg

File: pkg/grpc/metrics/grpc_metrics.go [hunk 2]
```diff
@@ -15,9 +16,22 @@ type GRPCMetrics interface {
...
+
 // NewGRPCMetrics returns a new GRPCMetrics object
 func NewGRPCMetrics() GRPCMetrics {
 	return &grpcMetricsImpl{
 		allMetrics: make(map[string]*perPathGRPCMetrics),
 	}
 }
+
+// GRPCSingleton returns a singleton of GRPCMetrics.
+func GRPCSingleton() GRPCMetrics {
```
Comment (connorgorman) [inline pkg/grpc/metrics/grpc_metrics.go:32]: Same comment as below, singletons should not be in pkg

**Review by connorgorman — CHANGES_REQUESTED** [review]
See comments about singletons

**Comment by RTann** [comment]
> See comments about singletons

I thought about that, but also didn't want to copy the singleton files over to Scanner since it will be exactly the same. Is that preferred over moving it into pkg?

**Comment by connorgorman** [comment]
So generally I would say it is preferred, but also I'm not sure these even need to be singletons. Typically the servers can be started from a single place so at that point why would we need to protect it with a once

**Comment by RTann** [comment]
Going to close this and just keep the singleton in Central. Will just not use the Central singleton in Scanner

**Pull request closed without merging**